src/TendoPay/Gateway.php

```diff
@@ -228,6 +238,10 @@ public function process_payment( $order_id ) {
 		$redirect_args = apply_filters( 'tendopay_process_payment_redirect_args_after_hash', $redirect_args, $order,
 			$this, $auth_token );
 
+		wc_reduce_stock_levels( $order->get_id() );
```

Comment:
This should not show up here. It has been moved to `src/TendoPay/TendoPay.php`. Please be sure to merge master properly, otherwise, you may introduce a critical bug like the one here.